syntax = 'proto3';

package lesson;

import "user.proto";

message Address {
  string name = 1;
}

message Audience {
  string name = 1;
  Address address = 2;
}

message LessonData {
  string id = 1;
  string name = 2;
}


message Attendance {
  AttendanceStatus status = 1;
  string decryption = 2;

  enum AttendanceStatus {
    PRESENT = 0;
    ABSENT = 1;
    SKIP_RESPECTFULLY = 2;
  }
}


message StudentAttendance {
  string student_id = 1;
  string full_name = 2;
  string decryption_of_full_name = 3;
  Attendance attendance = 4;
  string personal_number = 5;
  bool is_prefect = 6;
}


message GroupLesson {
  string id = 1;
  string complete_name = 2;
  repeated StudentAttendance attendances = 3;
  optional bool can_be_edited_by_prefect = 4;
}

message StudentLesson {
  Attendance attendance = 1;
  string group_id = 2;
  optional bool is_prefect = 3;
}

message TotalAttendance {
  int32 total_students = 1;
  int32 present_students = 2;
}

message Schedule {
  string id = 1;
  int32 number = 2;
  string date = 3;
  string type_of_lesson = 4;
  bool is_standardized = 5;
  optional int32 subgroup_number = 6;
  LessonData lesson = 7;
  optional StudentLesson student_data = 8;
  string time = 9;
  repeated Audience audiences = 10;
  repeated user.UserFullName teachers = 11;
  optional bool can_be_edited_by_prefect = 12;
  repeated string group_names = 13;
  optional TotalAttendance total_attendance = 14;
}